.github/workflows/_snap-release.yaml

```diff
@@ -0,0 +1,63 @@
+on:
+  workflow_call:
+    inputs:
+      channel:
+        description: |
+          The channel to which charm should be released. If specified (non-empty string),
+          the snap will be released to the input channel. If kept empty, the channel will
+          be auto-selected base on the current branch. E.g. if the current branch is
+          master or main, the snap will be released to latest/edge; otherwise, the snap will
+          be released to <branch-name>/edge.
+        default: ''
+        required: false
+        type: string
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Publish Snap
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: snap-artifact
+      - id: get-snap-name
+        run: echo "SNAP_FILE=$(ls *.snap)" >> $GITHUB_OUTPUT
+      - env:
+          REF: ${{ github.ref_name }}
+          CHANNEL: ${{ inputs.channel == '' }}
```

Comment:
Why there is `== ''`? The channel should be string and not bool value, or do I missing something here?

Reply:
This is a debugging step I used while testing. I have removed the block. Thanks for pointing it out.